File: src/main/java/de/maggiwuerze/xdccloader/util/FilesizeFormatter.java
package de.maggiwuerze.xdccloader.util;

import java.text.DecimalFormat;

public class FilesizeFormatter {

	public static String createAutoReadableString(long sizeInBytes) {

		String hrSize;

		double b = sizeInBytes;
		double k = sizeInBytes / 1024.0;
		double m = ((sizeInBytes / 1024.0) / 1024.0);
		double g = (((sizeInBytes / 1024.0) / 1024.0) / 1024.0);
		double t = ((((sizeInBytes / 1024.0) / 1024.0) / 1024.0) / 1024.0);

		DecimalFormat dec = new DecimalFormat("0.00");

		if (t > 1) {
			hrSize = dec.format(t).concat(" TB");
		} else if (g > 1) {
			hrSize = dec.format(g).concat(" GB");
		} else if (m > 1) {
			hrSize = dec.format(m).concat(" MB");
		} else if (k > 1) {
			hrSize = dec.format(k).concat(" KB");
		} else {
			hrSize = dec.format(b).concat(" Bytes");
		}

		return hrSize;
	}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/util/FilesizeFormatter.kt
package de.maggiwuerze.xdccwebloader.util

import java.text.DecimalFormat

object FilesizeFormatter {
    fun createAutoReadableString(sizeInBytes: Long): String {
        val hrSize: String?

        val b = sizeInBytes.toDouble()
        val k = sizeInBytes / 1024.0
        val m = ((sizeInBytes / 1024.0) / 1024.0)
        val g = (((sizeInBytes / 1024.0) / 1024.0) / 1024.0)
        val t = ((((sizeInBytes / 1024.0) / 1024.0) / 1024.0) / 1024.0)

        val dec = DecimalFormat("0.00")

        if (t > 1) {
            hrSize = "${dec.format(t)} TB"
        } else if (g > 1) {
            hrSize = "${dec.format(g)} GB"
        } else if (m > 1) {
            hrSize = "${dec.format(m)} MB"
        } else if (k > 1) {
            hrSize = "${dec.format(k)} KB"
        } else {
            hrSize = "${dec.format(b)} Bytes"
        }

        return hrSize
    }
}
